src/main/java/redis/clients/jedis/args/Endpoint.java

```diff
@@ -0,0 +1,69 @@
+package redis.clients.jedis.args;
+
+import redis.clients.jedis.Protocol;
+
+public interface Endpoint<T> extends Rawable {
+
+  Endpoint<T> inclusive();
+
+  Endpoint<T> exclusive();
+
+  public static abstract class AbstractEndpoint<T> implements Endpoint<T> {
+
+    private final byte[] raw;
+    private boolean exclusive;
+
+    public AbstractEndpoint(T data, boolean exclusive) {
+      if (data instanceof Rawable) {
+        this.raw = ((Rawable) data).getRaw();
+      } else if (data instanceof Integer) {
+        this.raw = Protocol.toByteArray((Integer) data);
+      } else if (data instanceof Double) {
+        this.raw = Protocol.toByteArray((Double) data);
+      } else {
+        throw new IllegalArgumentException(); // TODO message
+      }
+      this.exclusive = exclusive;
+    }
+
+    public AbstractEndpoint(T data) {
+      this(data, false);
+    }
+
+    @Override
+    public Endpoint<T> exclusive() {
```

Comment:
@sazzad16 why won't you update the raw on this call? instead of building each time on `getRaw()`

Reply:
@gkorland That's a reasonable request. Is this your only concern?

Reply:
yes, let's fix and merge this PR

Reply:
@gkorland Please take a look at #2546 to see my current approach to extend this PR.